services/app-web/src/pages/StateSubmission/StateSubmissionForm.test.tsx

```diff
@@ -566,4 +571,78 @@ describe('StateSubmissionForm', () => {
             expect(deleteCallKeys).toEqual(['three-one'])
         })
     })
+
+    describe('debounced form update', () => {
+        it('calls debounce when it calls the update', async () => {
+            const mockSubmission = mockDraftHealthPlanPackage({
+                id: '15',
+                submissionDescription: 'A real submission',
+            })
+            const debouncedSpy = jest.spyOn(useDebounce, 'useDebouncedMutation')
+            const mockUpdateSpy = jest.spyOn(
+                apolloMocks,
+                'updateHealthPlanFormDataMockSuccess'
+            )
+            const formData = base64ToDomain(
+                mockSubmission.revisions[0].node.formDataProto
+            )
+            if (formData instanceof Error) throw Error
+
+            const updatedFormData = domainToBase64(formData)
+
+            renderWithProviders(
+                <Routes>
+                    <Route
+                        path={RoutesRecord.SUBMISSIONS_FORM}
+                        element={<StateSubmissionForm />}
+                    />
+                </Routes>,
+                {
+                    apolloProvider: {
+                        mocks: [
+                            fetchCurrentUserMock({ statusCode: 200 }),
+                            fetchHealthPlanPackageMockSuccess({
+                                id: '15',
+                                submission: mockSubmission,
+                            }),
+                            updateHealthPlanFormDataMockSuccess({
+                                id: '15',
+                                pkg: mockSubmission,
+                                updatedFormData,
+                            }),
+                            fetchHealthPlanPackageMockSuccess({
+                                id: '15',
+                            }),
+                        ],
+                    },
+                    routerProvider: { route: '/submissions/15/edit/type' },
+                }
+            )
+
+            await waitFor(() =>
+                expect(
+                    screen.getByRole('form', { name: 'Submission Type Form' })
+                ).toBeInTheDocument()
+            )
+
+            const textarea = await screen.findByRole('textbox', {
+                name: 'Submission description',
+            })
+
+            const continueButton = await screen.findByRole('button', {
+                name: 'Continue',
+            })
+
+            void userEvent.type(textarea, 'updated something')
+
+            continueButton.click()
+            // looking for Step 2 of 6, which is a way to make sure the update has gone through and we're on the next page
+            await waitFor(() => {
+                expect(screen.getByText('2')).toBeInTheDocument()
+            })
+            expect(debouncedSpy).toHaveBeenCalled()
+            expect(mockUpdateSpy).toHaveBeenCalled()
+            jest.clearAllMocks()
```

Comment:
Not blocking but something I was looking at -- The test I would like to see is one where the user clicks multiple times and  we expect `mockUpdateSpy.toHaveBeenCalledTimes(1)`.

I messed around a bit with writing the test and using jest timers but then I couldn't get it failing when expected (e.g. I removed the hook and test still passed lol).

Reply:
This is exactly what I was talking about in the summary.  I spent a full day trying to get that test to work.  No luck.